src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -577,9 +581,11 @@ public String toPattern() {
                     result.append(",number,integer");
                 } else {
                     if (fmt instanceof DecimalFormat) {
-                        result.append(",number,").append(((DecimalFormat)fmt).toPattern());
+                        result.append(",number");
+                        subformatPattern = ((DecimalFormat)fmt).toPattern();
```

Comment:
Could use pattern matching `instanceof` here and in the other instances, but I understand if you're trying to stay consistent with the existing code.

Reply:
Agreed! Old habits :) I'll fix.